pkg/gofr/datasource/solr/solr_test.go

```diff
@@ -0,0 +1,178 @@
+package solr
+
+import (
+	"bytes"
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+)
+
+func Test_InvalidRequest(t *testing.T) {
+	_, err := call(context.TODO(), "GET", ":/localhost:", nil, nil)
+
+	require.Error(t, err, "TEST Failed.\n")
+}
+
+func Test_InvalidJSONBody(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		_, _ = w.Write([]byte(`Not a JSON`))
+	}))
+	defer ts.Close()
+
+	_, err := call(context.TODO(), "GET", ts.URL, nil, nil)
+
+	require.Error(t, err, "TEST Failed.\n")
+}
+
+func Test_ErrorResponse(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		http.Error(w, "some error", http.StatusLocked)
+	}))
+	ts.Close()
+
+	_, err := call(context.TODO(), "GET", ts.URL, nil, nil)
+
+	require.Error(t, err, "TEST Failed.\n")
+}
+
+func setupClient(t *testing.T) *Client {
+	t.Helper()
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		_, _ = w.Write([]byte(`{
+  		"responseHeader": {
+    	"rf": 1,
+    	"status": 0}}`))
+	}))
+
+	a := ts.Listener.Addr().String()
+	addr := strings.Split(a, ":")
+
+	ctrl := gomock.NewController(t)
+	mockLogger := NewMockLogger(ctrl)
+	mockMetrics := NewMockMetrics(ctrl)
+
+	mockLogger.EXPECT().Debug(gomock.Any())
+	mockMetrics.EXPECT().RecordHistogram(gomock.Any(), "app_solr_stats", gomock.Any(), "type", gomock.Any())
+
+	s := New(Config{Host: addr[0], Port: addr[1]})
+	s.metrics = mockMetrics
+	s.logger = mockLogger
+
+	return s
+}
+
+func Test_ClientSearch(t *testing.T) {
+	s := setupClient(t)
+
+	resp, err := s.Search(context.TODO(), "test", map[string]interface{}{"id": []string{"1234"}})
+
+	require.NoError(t, err, "TEST Failed.\n")
+	require.NotNil(t, resp, "TEST Failed.\n")
+}
+
+func Test_ClientCreate(t *testing.T) {
+	s := setupClient(t)
+
+	body := bytes.NewBuffer([]byte(`{
+		"id": "1234567",
+		"cat": [
+			"Book"
+		],
+		"genere_s": "Hello There"}`))
+
+	resp, err := s.Create(context.TODO(), "test", body, map[string]interface{}{"commit": "true"})
+
+	require.NoError(t, err, "TEST Failed.\n")
+	require.NotNil(t, resp, "TEST Failed.\n")
+}
+
+func Test_ClientUpdate(t *testing.T) {
+	s := setupClient(t)
+
+	body := bytes.NewBuffer([]byte(`{
+		"id": "1234567",
+		"cat": [
+			"Book"
+		]}`))
+	resp, err := s.Update(context.TODO(), "test", body, map[string]interface{}{"commit": "true"})
```

Comment:
```suggestion
	resp, err := s.Update(context.TODO(), "test", body, map[string]any{"commit": "true"})
```

But in fact, this should be OK


```suggestion
	resp, err := s.Update(context.TODO(), "test", body, map[string]string{"commit": "true"})
```